src/Contexts/Adaptors/SubstrateDestinationAdaptor.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useCallback, useEffect, useState } from "react";
```

Comment:
Please, move class splitting to a separate PR. It does not relate to the feature.

Reply:
Do we need to spend time on this? It's simpler to test everything in 1 PR so we spend less time on manual testing.